network-api/networkapi/wagtailpages/customblocks.py

```diff
@@ -443,23 +443,26 @@ class LatestProfileList(blocks.StructBlock):
         default=''
     )
 
-    def get_context(self, value, parent_context=None):
+    def get_context(self, value, parent_context=None, no_limit=False, initial_year=False):
         context = super().get_context(value, parent_context=parent_context)
+
         query_args = {
-            'limit': context['block'].value['max_number_of_results'],
-            'profile_type': context['block'].value['profile_type'],
-            'program_type': context['block'].value['program_type'],
-            'program_year': context['block'].value['year'],
+            'limit': value['max_number_of_results'],
+            'profile_type': value['profile_type'],
+            'program_type': value['program_type'],
+            'program_year': initial_year if initial_year else value['year'],
```

Comment:
Having this parameter if we can _only_ filter by year is misleading

Reply:
JK, I get it

Reply:
No it isn't: this is the super class, which absolutely needs to respect its own `year` value. `initial_year` only gets used if the subclass passes that along through its `super.get_context` call.